### PR TITLE
Add a warning to the system semantic conventions

### PR DESCRIPTION
**Pull request by dmitryax**
Add a warning to the system semantic conventions to prevent implementations from adopting the breaking changes until the conventions are stabilized. This warning is to be updated once https://github.com/open-telemetry/semantic-conventions/issues/246 is closed.

As discussed in the last system conventions meeting.

cc @jsuereth @open-telemetry/semconv-system-approvers

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [ ] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

File: docs/system/hardware-metrics.md [hunk 1]
```diff
@@ -33,6 +33,18 @@ when creating instruments not explicitly defined in the specification.
 
 <!-- tocstop -->
 
+> **Warning**
+> Existing instrumentations and collector that are using
+> [v1.21.0 of this document](/../v1.21.0/docs/system/hardware-metrics.md)
+> (or prior):
+>
+> * SHOULD NOT adopt any breaking changes from the future versions of this
+>   document system until the system semantic conventions are marked stable.
```
Comment (ChrsMark) [inline docs/system/hardware-metrics.md:42]:
```suggestion
>   document until the system semantic conventions are marked stable.
```
I'm not sure what this extra `system` stands for 🤔

Comment (dmitryax) [reply docs/system/hardware-metrics.md]: Good catch. Thanks :)

**Review by ChrsMark — APPROVED** [review]
Content wise LGTM

File: docs/system/hardware-metrics.md [hunk 2]
```diff
@@ -33,6 +33,18 @@ when creating instruments not explicitly defined in the specification.
 
 <!-- tocstop -->
 
+> **Warning**
+> Existing instrumentations and collector that are using
+> [v1.21.0 of this document](/../v1.21.0/docs/system/hardware-metrics.md)
+> (or prior):
+>
+> * SHOULD NOT adopt any breaking changes from the future versions of this
+>   document until the system semantic conventions are marked stable.
```
Comment (trask) [inline docs/system/hardware-metrics.md:42]: ("future versions" initially read to me like versions in the future from now, but not necessarily including the current version)
```suggestion
> * SHOULD NOT adopt any breaking changes from this
>   document until the system semantic conventions are marked stable.
```

**Comment by dmitryax** [comment]
The test is fixed in https://github.com/open-telemetry/semantic-conventions/pull/583

**Pull request merged**